packages/wdio-reporter/src/stats/suite.ts

```diff
@@ -0,0 +1,48 @@
+import HookStats from './hook'
+import RunnableStats, { Runnable } from './runnable'
+import TestStats from './test'
+
+export interface Suite extends Runnable {
+    type?: string
+    cid?: string
+    title: string
+    fullTitle: string
+    tags?: string[]
+    tests?: TestStats[]
+    hooks?: HookStats[]
+    suites?: SuiteStats[]
+    hooksAndTests?: (HookStats | TestStats)[]
+    description?: string | undefined
+}
+
+/**
+ * Class describing statistics about a single suite.
+ */
+export default class SuiteStats extends RunnableStats {
+    uid: string
+    cid?: string
+    title: string
+    fullTitle: string
+    tags?: string[]
+    tests: TestStats[] = []
+    hooks: HookStats[] = []
+    suites: SuiteStats[] = []
+    /**
+     * an array of hooks and tests stored in order as they happen
+     */
+    hooksAndTests: (HookStats | TestStats)[] = []
+    description?: string
+
+    constructor (suite: Suite) {
+        super(suite.type || 'suite')
```

Comment:
This prevents us from using a string literal as the type in Runner. Any idea what values suite.type can be?

Reply:
The type has been assigned in the Suite interface like so: type: string | 'suite:start' | 'suite:end'
Hence I think you cannot simply reassign it only a string type.